src/sentry/static/sentry/app/views/groupDetails.jsx

```diff
@@ -48,13 +50,42 @@ const GroupDetails = React.createClass({
     this.fetchData();
   },
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.params.groupId !== this.props.params.groupId) {
+      this.remountComponent();
+    }
+  },
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevProps.params.groupId !== this.props.params.groupId) {
+      this.fetchData();
+    }
+  },
+
   remountComponent() {
-    this.setState(this.getInitialState(), this.fetchData);
+    this.setState(this.getInitialState());
```

Comment:
This follows the pattern from [ProjectDetails](https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/projectDetails.jsx), but I'm not totally  sure this actually works with `LoadingError` on L147 below.